### PR TITLE
SOC-5928 Member shouldn't be able to change Space Banner

### PR DESCRIPTION
**Pull request by boubaker**

### CODE REVIEW

File: component/webui/src/main/java/org/exoplatform/social/webui/space/UISpaceMenu.java [hunk 1]
```diff
@@ -181,6 +193,12 @@ public void execute(Event<UISpaceMenu> event) throws Exception {
     }
   }
 
+  public boolean isManager() {
```
Comment (thomasdelhomenie) [inline component/webui/src/main/java/org/exoplatform/social/webui/space/UISpaceMenu.java]: Should be renamed to canEditBanner or something like IMO.

File: extension/war/src/main/webapp/groovy/social/webui/space/UISpaceMenu.gtmpl [hunk 2]
```diff
@@ -104,9 +110,11 @@ if(appSelected == null || appSelected.length() == 0) {
 <ul class="spaceMenuApps menuApps pull-left">
 </ul>
         <div class="bannerControls">
-          <a href="#"><i class="uiIconCamera"></i></a>
-          <% if (banner != null) {%>
-          <a href="<%=uicomponent.event("DeleteBanner");%>"><i class="uiIconTrash"></i></a>
+          <%if(uicomponent.isManager()) {%>
```
Comment (thomasdelhomenie) [inline extension/war/src/main/webapp/groovy/social/webui/space/UISpaceMenu.gtmpl]: Why not removing the div bannerControls when we do not have edit permissions ?

File: component/webui/src/main/java/org/exoplatform/social/webui/space/UISpaceMenu.java (hunk 1 above)
Comment (boubaker) [reply component/webui/src/main/java/org/exoplatform/social/webui/space/UISpaceMenu.java]: done

File: extension/war/src/main/webapp/groovy/social/webui/space/UISpaceMenu.gtmpl (hunk 2 above)
Comment (boubaker) [reply extension/war/src/main/webapp/groovy/social/webui/space/UISpaceMenu.gtmpl]: done

**Comment by boubaker** [comment]
> Did you check the permissions check is also done at service level ?

The API don't allow it. BTW, multiple services aren't aware of this and thus it should be considered everywhere. For this fix, I will attempt to change API, but this shouldn't be a breaking change, so I will deprecate the old methods (since eXo Platform 4) and define a new ones.
WDYT ?

**Comment by thomasdelhomenie** [comment]
If the API you are talking about is the methods related to space and profile banner,  they have been introduced in 5.0, so no need for deprecation. If you are talking of others APIs, I agree.

**Comment by boubaker** [comment]
> others APIs

yes, especially updateSpace.

As discussed, I will use editor field from Space entity to do this check for now. But, it should be more explicit.

**Comment by boubaker** [comment]
done

**Comment by boubaker** [comment]
BTW, we couldn't apply this for IdentityManager API, the changes will be more important IMO

**Pull request merged**